Add acceptance/selftest, showcasing basic features

denik opened the pull request:
Also make TestInprocessMode use this test.

pietern reviewed (approved):
Can you update `acceptance/README.md` to point at this?

It would be nice to include custom test configuration as well.

denik commented:
> Can you update acceptance/README.md to point at this?

Updated

> It would be nice to include custom test configuration as well.

I've added test.toml with all supported fields (commented out), would be good to use this also as a demo of test.toml.
